app/models/case_cycle.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# The cycle for subsidy cases
+class CaseCycle < UuidApplicationRecord
+  monetize :copay_cents
+
+  enum copay_frequency: { daily: 'daily', weekly: 'weekly', monthly: 'monthly' }
+  enum status: { submitted: 'submitted', pending: 'pending', approved: 'approved', denied: 'denied' }
+
+  before_save :set_slug
```

Comment:
FWIW, I like this style better than using a lambda/block here.  Is cleaner to me.